pkg/client/database/timescale/timescale.go

```diff
@@ -97,3 +113,39 @@ func (e Executor) Finalize(ctx context.Context, retErr error) error {
 
 	return retErr // keep migrate.ErrNoChange
 }
+
+func runningTimescaleVersion(ctx context.Context, q queryRower) (*version.Version, error) {
+	row := q.QueryRow(ctx, timescaleVersionQuery)
+	var result string
+	if err := row.Scan(&result); err != nil {
+		return nil, err
+	}
+	fields := strings.Fields(result)
+	if len(fields) != 1 {
+		return nil, fmt.Errorf("failed to parse version: %s, %q", result, fields)
+	}
+	return version.NewVersion(fields[0])
+}
+
+// ValidateTimescaleVersion checks that Timescale plugin version available through pool is not lower than wanted version.
+// In this case it returns nil. Otherwise returns error describing current and desired version or any other error encountered
+// during the check.
+func ValidateTimescaleVersion(ctx context.Context, pool *pgxpool.Pool, want *version.Version) error {
```

Comment:
I would use the const `MinTimescaleVersion` here and not require `want`. (in fact the same logic applies to `ValidatePostgresVersion`)

Reply:
You can't make it a const:
```
version.Must(version.NewVersion("2.0")) (value of type *version.Version) is not constant 
```

Reply:
no use the const in the top line with the string.

Reply:
ok saw the edit, `var` is also ok. but my point was, to make it like:
`func ValidateTimescaleVersion(ctx context.Context, pool *pgxpool.Pool) error`

ValidateTimescaleVersion uses the const inside of the fn, caller doesn't need to specify or know about what version or what const is required.

Reply:
This change would require changes to the tests

Reply:
Tests would use `doValidateTimescaleVersion`?